aip/0203.md

```diff
@@ -119,10 +119,17 @@ field on a request message (a message that is an argument to an RPC, usually
 ending in `Request`), or a field of a message included within a request
 message.
 
+When a service receives an immutable field in an update request (or similar),
+even if included in the update mask, the service **should** ignore the field
+and leave the existing value alone, but **should not** error.
+
```

Comment:
Is this saying that it is never an error to try to change an immutable field using an update, but that it just doesn't do anything?  If so, that seems confusing to users to debug what is going on.

Or is it saying something more subtle?

I would have expected it to say something like:

_If an update request asks to change to an immutable field, then it *should* return an error.  A change is defined as the value in the update differs from the value in the current state of the resources on the server, and that the field was included in the update mask._
_If an update request mentions an immutable field but does not ask to change it, this *should* be ignored._

Reply:
> If an update request asks to change to an immutable field, then it should return an error. A change is defined as the value in the update differs from the value in the current state of the resources on the server, and that the field was included in the update mask.
> If an update request mentions an immutable field but does not ask to change it, this should be ignored.

I just realized I was thinking of output only, and not immutable, when I wrote the rule. For output only, we ignore all the time (because of things like update_time), but I agree that for immutable, we should error if a change is requested. Updating.